source/common/redis/conn_pool_impl.cc

```diff
@@ -125,6 +131,7 @@ void ClientImpl::onRespValue(RespValuePtr&& value) {
   ASSERT(!pending_requests_.empty());
   PendingRequest& request = pending_requests_.front();
   if (!request.canceled_) {
+    host_->outlierDetector().putHttpResponseCode(enumToInt(Http::Code::OK));
```

Comment:
I see your point. IMO I would probably still change it to also include cancels, until we have more robust timeout/stats as you point out, since I think it is technically correct.